colrev/ops/built_in/search_sources/ieee.py

```diff
@@ -79,28 +88,167 @@ def heuristic(cls, filename: Path, data: str) -> dict:
 
         result = {"confidence": 0.1}
 
-        if "INPROCEEDINGS" in data:
-            if len(re.findall(r"@[A-Z]*\{[0-9]*,\n", data)) >= data.count("\n@"):
-                result["confidence"] = 1.0
-        if all(
-            x in data.splitlines()[0] for x in ["Date Added To Xplore", "IEEE Terms"]
-        ):
-            result["confidence"] = 1.0
-
         return result
 
     @classmethod
     def add_endpoint(
         cls, search_operation: colrev.ops.search.Search, query: str
     ) -> colrev.settings.SearchSource:
         """Add SearchSource as an endpoint (based on query provided to colrev search -a )"""
+        if "https://ieeexploreapi.ieee.org/api/v1/search/articles?" in query:
+            query = (query.replace("https://ieeexploreapi.ieee.org/api/v1/search/articles?", "").lstrip("&")
+            )            
+
+            parameter_pairs = query.split("&")
+            search_parameters = {}
+            for parameter in parameter_pairs:
+                key, value = parameter.split("=")
+                search_parameters[key] = value
+
+            last_value = list(search_parameters.values())[-1]
+
+            filename = search_operation.get_unique_filename(
+                file_path_string=f"ieee_{last_value}"
+            )
+
+            add_source = colrev.settings.SearchSource(
+                endpoint="colrev.ieee",
+                filename=filename,
+                search_type=colrev.settings.SearchType.DB,
+                search_parameters=search_parameters,
+                load_conversion_package_endpoint={"endpoint": "colrev.bibtex"},
+                comment="",
+            )
+            return add_source
+          
         raise NotImplementedError
 
     def run_search(
         self, search_operation: colrev.ops.search.Search, rerun: bool
     ) -> None:
         """Run a search of IEEEXplore"""
 
+        ieee_feed = self.search_source.get_feed(
+            review_manager=search_operation.review_manager,
+            source_identifier=self.source_identifier,
+            update_only=(not rerun),
+        )   
+        settings=self.review_manager.environment_manager.load_environment_registry()
+        #if statement is used so that the api_key does not need to be entered every time after first insertion. But is this a proper validation of the key?
+        if len(settings ['packages']) != 24:
```

Comment:
They api_key should be stored in `packages.search_source.colrev.ieee.api_key`. The `environment_manager.get_settings_by_key()` method makes it easier to handle nested dictionaries (as in the unpaywall example).